backend/fetchArrivals.ts

```
@@ -1,18 +1,65 @@
 import axios from 'axios';
+import * as dfns from 'date-fns';
 
 const key = import.meta.env.VITE_API_KEY;
 
-export default async function getArrivals(id: string){
+export class Bus {
+    lineName: string;
+    destinationName : string;
+    towards: string;
+    expectedArrival: Date;
+    timeToStation: number
+
+    constructor(lineName = "Not given", destinationName = "Not given", towards = "Not given", expectedArrival = "2025-01-01T00:00:00Z", timeToStation = 1440) {
```

do you need defaults for all the values?

---

This was intended to handle the possibility of missing data and avoid null / undefined handling down the line (with arrayDto: success = False handling complete failures of the api). Is it possible to mark properties as nullable in ts and is that better practice here?

---

Yep - you can use the `linename: string?` syntax.

Thats probably a better option, since you can handle it down the line & assume if you've got a string then it must be  valid. This will avoid things failing silently, or the error messages occuring far away from the source